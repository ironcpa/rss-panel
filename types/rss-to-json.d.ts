declare module 'rss-to-json';
