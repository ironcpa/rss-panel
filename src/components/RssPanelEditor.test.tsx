import React from 'react';
import { shallow } from 'enzyme';
import { RssPanelEditor } from './RssPanelEditor';
import { PanelEditorProps } from '@grafana/ui';
import { RssOptions } from '../types';

const setup = (propOverrides?: Partial<PanelEditorProps<RssOptions>>) => {
  const props: PanelEditorProps<RssOptions> = {
    options: {
      feedUrl: 'some/url',
    },
    onOptionsChange: jest.fn(),
  };

  Object.assign(props, propOverrides);

  return shallow(<RssPanelEditor {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
