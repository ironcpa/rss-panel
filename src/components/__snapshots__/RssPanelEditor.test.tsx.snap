// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<DocumentFragment>
  <div
    class="panel-options-group"
  >
    <div
      class="panel-options-group__header"
    >
      <span
        class="panel-options-group__title"
      >
        Feed
      </span>
    </div>
    <div
      class="panel-options-group__body"
    >
      <div
        class="gf-form"
      >
        <div
          class="form-field"
        >
          <label
            class="gf-form-label width-6"
          >
            Feed url
          </label>
          <input
            class="gf-form-input width-25"
            type="text"
            value="some/url"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
